lib/karafka/processing/jobs/base.rb

```diff
@@ -22,9 +22,11 @@ def initialize
           @non_blocking = false
         end
 
-        # When redefined can run any code prior to the job being enqueued
+        # When redefined can run any code prior to the job being scheduled
         # @note This will run in the listener thread and not in the worker
-        def before_schedule; end
+        def before_schedule
+          raise NotImplementedError, 'Please implement in a subclass'
```

Comment:
`Implement in a subclass`

Reply:
Would have to change in all the places where this message is. Outside of the scope but feel free to create an issue :)

Reply:
Fair enough.